Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+    {
+        RelatedRecipes = content 
+            .Parent           
+            .Children
+            .Where(x => x.GetPropertyValue<IEnumerable<int>>("related")
+                            .Contains(content.Id));
+
+        Votes = content.GetPropertyValue<int>("votes");
+    }
+
+    public int Votes { get; private set; }
+    public IEnumerable<RecipeModel> RelatedRecipes { get; private set; }
+}
+```
+* You then run the following code to show to show the favorites 
+```
+@var recipeNode = Umbraco.TypedContent(3251);
+<ul>
+@foreach(var recipe in recipeNode.Children
+                            .Select(x => new RecipeModel(x))
+                            .OrderByDescending(x => x.Votes)
+                            .Take(10))
+{
+    <li><a href="@recipe.Url">@recipe.Name</a></li>    
+}
+</ul>
+```
+
+__Ouch!__ So just to show the top 10 voted recipe's this will end up doing the following:
+
+* This will iterate over all Recipess, create and allocate 5000 instances of `IPublishedContent`
+* This will create and allocate 5000 instances of `RecipeModel`
+* For each `RecipeModel` created, this will traverse upwards, iterate all 5000 recipes then resolve property data for 2 properties
+
+This means that there is now a minimum of __20,000__ new objects created and allocated in memory. The number of traversals/visits to each
+of these objects is now 5000 x 5000 = __25,000,000 (25 MILLION TIMES!)__
+
+_Side note: The other problem is the logic used to lookup related recipes is incredibly inneficient. Instead, each reciple
+should have a picker to choose it's related recipe's and then each of those can just be looked up by their ID.
+(There's probably a few other ways to achieve this too!)_
+
+Which leads us on to the next anti-pattern...
+
+## Don't eager load data, lazy load it instead
+
+The above example could be rewritten like this:
+
+```csharp
+public class RecipeModel : PublishedContentWrapped
+{
+    public RecipeModel(IPublishedContent content) : base(content)
+    {
+    }
+
+    private int? _votes;
+    public int Votes 
+    {
+        get 
+        {
+            //Lazy load the property value and ensure it's not re-resolved once it's loaded
+            return _votes ?? (_votes = GetPropertyValue<int>("votes"));
+        } 
+    }
+
+    //Just return the Ids, they can be resolved to IPublishedContent instances in the view or elsewhere,
+    // doesn't need to be in the model - this would also be bad if the model was cached since all of the
+    // related entities would end up in the cache too.
+    private List<int> _related;
+    public IEnumerable<int> RelatedRecipes
+    {
+        get 
+        {
+            //Lazy load the property value and ensure it's not re-resolved once it's loaded            
+            return _related ?? 
+                (_related = GetPropertyValue<IEnumerable<int>>("related").ToList());
+        } 
+    }    
+}
+```
+
+This is slightly better:
```

Comment:
Good to see the bad example & improved example, but this copy of `this is slightly better` indicates to me that this could be improved more, if so then why not expand on it further to the best scenario?